src/masternodes/mn_checks.cpp

```diff
@@ -2088,6 +2094,11 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
         return mnview.StoreVault(obj.vaultId, *vault.val);
     }
 
+
+    Res operator()(const CDepositToVaultMessage& obj) const {
+        return mnview.AddVaultCollateral(obj.vaultId, obj.amount);
```

Comment:
Need to check the auth of the from address, check they have enough balance to cover the amount and I presume the deduct amount from their balance.

Reply:
Working on this right now!

Reply:
Also if no collateral first should be DFI, after adding should check against remains 50% collateral of DFI. Also vault can be in liquidation state i.e. auction is started.